drivers/usb/gadget/function/f_scm.c

```diff
@@ -0,0 +1,574 @@
+
+	.bNumEndpoints      = 4,
+	.bInterfaceClass    = USB_CLASS_VENDOR_SPEC,
+	.bInterfaceSubClass = SCM_SUBCLASS,
+};
+
+/***************************************************************************
+ * USB DESCRIPTOR DEFINITIONS
+ * There are 4 descriptors:
+ *   Bulk In / Out
+ *   Cmd In / Out
+ * There are 3 speeds:
+ *   Full Speed
+ *   High Speed
+ *   Super Speed
+ * Every combination of the above needs its own descriptor.
+ ***************************************************************************/
+
+/**
+ * Full speed endpoint descriptors
+ */
+static struct usb_endpoint_descriptor
+fs_scm_cmd_in_desc = {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_OUT,
+	.bmAttributes =     USB_ENDPOINT_XFER_INT,
+	.wMaxPacketSize =   cpu_to_le16(MAX_INT_PACKET_SIZE),
+	.bInterval =        SCM_STATUS_INTERVAL_MS,
+};
+
+static struct usb_endpoint_descriptor
+fs_scm_cmd_out_desc = {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_IN,
+	.bmAttributes =     USB_ENDPOINT_XFER_INT,
+	.wMaxPacketSize =   cpu_to_le16(MAX_INT_PACKET_SIZE),
+	.bInterval =        SCM_STATUS_INTERVAL_MS,
+};
+
+static struct usb_endpoint_descriptor fs_scm_in_desc = {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_OUT,
+	.bmAttributes =     USB_ENDPOINT_XFER_BULK,
+
+};
+
+static struct usb_endpoint_descriptor fs_scm_out_desc =  {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_IN,
+	.bmAttributes =     USB_ENDPOINT_XFER_BULK,
+
+};
+
+static struct usb_descriptor_header *scm_fs_descs[] = {
+	(struct usb_descriptor_header *) &scm_intf,
+	(struct usb_descriptor_header *) &fs_scm_in_desc,
+	(struct usb_descriptor_header *) &fs_scm_out_desc,
+	(struct usb_descriptor_header *) &fs_scm_cmd_in_desc,
+	(struct usb_descriptor_header *) &fs_scm_cmd_out_desc,
+	NULL,
+};
+
+/**
+ * High speed descriptors
+ */
+static struct usb_endpoint_descriptor
+hs_scm_cmd_in_desc = {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_OUT,
+	.bmAttributes =     USB_ENDPOINT_XFER_INT,
+	.wMaxPacketSize =   cpu_to_le16(MAX_INT_PACKET_SIZE),
+	.bInterval =        USB_MS_TO_HS_INTERVAL(SCM_STATUS_INTERVAL_MS),
+};
+
+static struct usb_endpoint_descriptor
+hs_scm_cmd_out_desc = {
+	.bLength =          USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =  USB_DT_ENDPOINT,
+
+	.bEndpointAddress = USB_DIR_IN,
+	.bmAttributes =     USB_ENDPOINT_XFER_INT,
+	.wMaxPacketSize =   cpu_to_le16(MAX_INT_PACKET_SIZE),
+	.bInterval =        USB_MS_TO_HS_INTERVAL(SCM_STATUS_INTERVAL_MS),
+};
+
+static struct usb_endpoint_descriptor hs_scm_in_desc = {
+	.bLength =              USB_DT_ENDPOINT_SIZE,
+	.bDescriptorType =      USB_DT_ENDPOINT,
+
+	.bmAttributes =         USB_ENDPOINT_XFER_BULK,
```

Comment:
Set `.bEndpointAddress` on the high-speed and super-speed in&out bulk endpoints descriptors as well.  You already do for the interrupt command descriptors.

I know it isn't strictly necessary, since those values (with the dynamic portion added) are copied from the fullspeed variant in `bind`.  But including the endpointAddress direction here improves code readability (which is why `f_emc.c` does it).